src/consensus/pbft/libbyz/Replica.cpp

```diff
@@ -1847,10 +1853,14 @@ void Replica::execute_prepared(bool committed)
       }
     }
 
-    if (global_commit_cb != nullptr)
+    if (global_commit_cb != nullptr && pp->is_signed())
```

Comment:
Is this really required here? I thought it would be safe to global commit if we get this far, whether it is a signed batch or not.

Reply:
as I understand it without this we will globally commit on every transaction, where as with this we globally commit only when we sign (similar to raft)

Reply:
This is correct we do not want to sign every batch, that is computationally expensive